### PR TITLE
 spotify-unwrapped: 20.09 backport: 1.1.26.501 -> 1.1.55.498

### PR DESCRIPTION
**Pull request by sheenobu**
…a83c60

backport the changes in master to 20.09.
includes:

 * ffmpeg4 migration (#120705)
 * rename to spotify-unwrapped
 * high dpi override support
 * gtk2->gtk3
 * 20.09 specific: curlWithGnuTls

###### Motivation for this change

Mostly #120705 . 

NOTE: I am not sure how I feel about renaming this to spotify-unwrapped in 20.09. This will break peoples configs if they have nonFreeFilter set to "spotify" instead of spotify-unwrapped. Those types of changes are best for major upgrades IMO. 

The rest of the changes I'm OK with.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121553` at a82afb32 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>spotify</li>
    <li>spotifywm</li>
  </ul>
</details>

**Comment by Ma27** [comment]
Is there a **good** reason for the breaking change (i.e. `-unwrapped`)? If not, then we should leave it out.

**Comment by jonringer** [comment]
I only see the `pname` being changed, so the change is probably zero for most users. But I think we should leave that out as it's not super necessary for 20.09.

File: pkgs/applications/audio/spotify/default.nix [hunk 1]
```diff
@@ -64,7 +69,7 @@ let
 in
 
 stdenv.mkDerivation {
-  pname = "spotify";
+  pname = "spotify-unwrapped";
```
Comment (yurrriq) [inline pkgs/applications/audio/spotify/default.nix:72]:
```suggestion
  pname = "spotify";
```

**Comment by mweinelt** [comment]
Unfortunately NixOS 20.09 has reached its end-of-life status on 2021-07-01, one
month after the release of NixOS 21.05.

Since we do not accept any changes to its branches anymore, I'm closing this pull
request.

**Pull request closed without merging**